### PR TITLE
rgw_file: add simple support for r/w open

### PR DESCRIPTION
**Pull request by Wilhelmshaven**
Need your advice for this, tkx :)

Signed-off-by: lihongjie <lihongjie@cmss.chinamobile.com>

### CODE REVIEW

File: src/test/librgw_file_aw.cc [hunk 1]
```diff
@@ -260,6 +260,36 @@ TEST(LibRGW, CLOSE2) {
...
+  ASSERT_EQ(ret, 0);
+}
+
+TEST(LibRGW, OPEN4) {
+  int ret = rgw_open(fs, object_fh, 0 /* posix flags */, RGW_OPEN_FLAG_WRITE);
+  ASSERT_EQ(ret, 0);
+}
+
+TEST(LibRGW, CLOSE4) {
+  int ret = rgw_close(fs, object_fh, RGW_OPEN_FLAG_WRITE);
```
Comment (mattbenjamin) [inline src/test/librgw_file_aw.cc]: this looks suspicious--need to pass an open flag to close?

File: src/test/librgw_file_aw.cc [hunk 2]
```diff
@@ -260,6 +260,36 @@ TEST(LibRGW, CLOSE2) {
...
+  ASSERT_EQ(ret, 0);
+}
+
+TEST(LibRGW, OPEN5) {
+  int ret = rgw_open(fs, object_fh, 0 /* posix flags */, RGW_OPEN_FLAG_WRITE | RGW_OPEN_FLAG_READ);
+  ASSERT_EQ(ret, 0);
+}
+
+TEST(LibRGW, CLOSE5) {
+  int ret = rgw_close(fs, object_fh, RGW_OPEN_FLAG_WRITE | RGW_OPEN_FLAG_READ);
```
Comment (mattbenjamin) [inline src/test/librgw_file_aw.cc]: ditto here

**Comment by mattbenjamin** [comment]
@Wilhelmshaven agree we'll need read and write opens at some point;  for writes currently, we can only support one writer;  I'm confused why we would pass read or write open flags to the close operation;  can you elaborate on that?

File: src/test/librgw_file_aw.cc (hunk 1 above)
Comment (Wilhelmshaven) [inline src/test/librgw_file_aw.cc]: Oops, sorry, my fault, I pasted the code on the wrong place...

**Comment by mattbenjamin** [comment]
@Wilhelmshaven continue discussion?

**Comment by Wilhelmshaven** [comment]
@mattbenjamin  Agree that only one writer shoule be supported. 
As you can see, instead of the original exclusive open, these two flags would implement two scenarios for now, one is exclusive write open, the other one is multiple read-only open. For more scenario such as 1-write-open + multi-read-only-open, I'm not sure whether it is right to do that.

**Comment by mattbenjamin** [comment]
@Wilhelmshaven ok, let's merge this after L branches

**Comment by Wilhelmshaven** [comment]
Conflict resolved.

**Comment by Wilhelmshaven** [comment]
retest this please

**Comment by mattbenjamin** [comment]
@Wilhelmshaven thanks for reopening

**Pull request closed without merging**